ext/spl/spl_array.c

```diff
@@ -1878,6 +1890,24 @@ SPL_METHOD(Array, __unserialize)
 	}
 
 	object_properties_load(&intern->std, Z_ARRVAL_P(members_zv));
+
+	if (iterator_class_zv && Z_TYPE_P(iterator_class_zv) != IS_NULL) {
```

Comment:
We must not assume that `iterator_class_zv` holds a string (next line), unless that has been verified. So
```suggestion
	if (iterator_class_zv && Z_TYPE_P(iterator_class_zv) == IS_STRING) {
```

Reply:
It has actually been verified, because if it is not a string, the function will return a few lines up. But your code is better anyways. Thank you!